narwhals/_pandas_like/dataframe.py

```diff
@@ -268,15 +269,52 @@ def join(
         self,
         other: Self,
         *,
-        how: Literal["left", "inner", "outer"] = "inner",
-        left_on: str | list[str],
-        right_on: str | list[str],
+        how: Literal["left", "inner", "outer", "cross"] = "inner",
+        left_on: str | list[str] | None = None,
+        right_on: str | list[str] | None = None,
     ) -> Self:
         if isinstance(left_on, str):
             left_on = [left_on]
         if isinstance(right_on, str):
             right_on = [right_on]
 
+        if how == "cross":
+            if self._implementation in {"modin", "cudf"} or (
+                self._implementation == "pandas"
+                and (pd := get_pandas()) is not None
+                and parse_version(pd.__version__) < parse_version("1.2.0")
+            ):
+
+                def generate_unique_token(
+                    n_bytes: int, columns: list[str]
+                ) -> str:  # pragma: no cover
+                    import secrets
+
+                    while True:
+                        token = secrets.token_hex(n_bytes)
+                        if token not in columns:
+                            return token
```

Comment:
this should be unlikely/impossible, but still, could we raise an assertion error if for some reason the loop doesn't exit after 100 iterations?